### PR TITLE
Allow to append logs after a post processing hook

### PR DESCRIPTION
**Pull request by thomas-nguy**
Closes: #1024

## Description

Add the ability to append log to the receipt after a tx post processing hook 

more information on the issue ticket.

open for discussion

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by fedekunze — CHANGES_REQUESTED** [review]
Can't you just modify the logs from the receipt?

**Comment by thomas-nguy** [comment]
Yes, but the computation of the bloom needs to be moved down nevertheless.

Do you prefer to have this ability implicitly and keep the previous signature?

**Comment by fedekunze** [comment]
> Yes, but the computation of the bloom needs to be moved down nevertheless.

Why? if you are updating the logs then you should also be updating the bloom during the post tx processing step. We'd need to change the state transition to do the following instead:

```go
	if len(receipt.Logs) > 0 {
		// Update transient block bloom filter
		k.SetBlockBloomTransient(ctx, receipt.Bloom)
		k.SetLogSizeTransient(ctx, uint64(txConfig.LogIndex)+uint64(len(receipt.Logs)))
	}
```

**Comment by thomas-nguy** [comment]
okay, make sense. Published a minimal change version

File: x/evm/keeper/state_transition.go [hunk 1]
```diff
@@ -282,9 +282,8 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, tx *ethtypes.Transaction) (*t
 
 	if len(logs) > 0 {
```
Comment (fedekunze) [inline x/evm/keeper/state_transition.go:283]:
```suggestion
	if len(receipt.Logs) > 0 {
```

**Review by fedekunze — APPROVED** [review]
LGTM

**Pull request merged**